lms/djangoapps/lti_provider/tasks.py

```diff
@@ -44,55 +53,80 @@ def score_changed_handler(sender, **kwargs):  # pylint: disable=unused-argument
         )
 
 
+def increment_assignment_versions(course_key, usage_key, user_id):
+    """
+    Update the version numbers for all assignments that are affected by a score
+    change event. Returns a list of all affected assignments.
+    """
+    problem_descriptor = modulestore().get_item(usage_key)
+    # Get all assignments involving the current problem for which the campus LMS
+    # is expecting a grade. There may be many possible graded assignments, if
+    # a problem has been added several times to a course at different
+    # granularities (such as the unit or the vertical).
+    assignments = outcomes.get_assignments_for_problem(
+        problem_descriptor, user_id, course_key
+    )
+    for assignment in assignments:
+        assignment.version_number += 1
+        assignment.save()
+    return assignments
+
+
 @CELERY_APP.task
-def send_outcome(points_possible, points_earned, user_id, course_id, usage_id):
+def send_composite_outcome(user_id, course_id, assignment_id, version):
     """
-    Calculate the score for a given user in a problem and send it to the
-    appropriate LTI consumer's outcome service.
+    Calculate and transmit the score for a composite module (such as a
+    vertical). A composite module may contain multiple problems, so we need to
+    calculate the total points earned and possible for all child problems. This
+    requires calculating the scores for the whole course, which is an expensive
+    operation.
```

Comment:
Nit: Can you add blank lines between paragraphs for readability?

Also, I would recommend leaving the first sentence in its own paragraph to be a short easy-to-read summary of the function.